### PR TITLE
Add tokenflow button

### PR DESCRIPTION
**Pull request by MightyDevKnight**
image

### CODE REVIEW

File: src/app/components/EditTokenForm.tsx [hunk 1]
```diff
@@ -172,28 +175,35 @@ function EditTokenForm({ resolvedTokens }: Props) {
...
-  const handleDescriptionChange = React.useCallback(
-    (val: string) => {
+  const handleDescriptionChange = React.useCallback<React.ChangeEventHandler<HTMLInputElement>>(
+    (e) => {
+      e.persist();
       if (internalEditToken) {
         setInternalEditToken({
           ...internalEditToken,
-          description: val,
+          description: e.target.value,
```
Comment (six7) [inline src/app/components/EditTokenForm.tsx:206]: Why was this change required?

File: src/app/components/Navbar.tsx [hunk 2]
```diff
@@ -8,17 +10,51 @@ import { NavbarUndoButton } from './NavbarUndoButton';
...
   const dispatch = useDispatch<Dispatch>();
   const { handleResize } = useMinimizeWindow();
+  const activeTheme = useSelector(activeThemeSelector);
+  const availableThemes = useSelector(themeOptionsSelector);
+  const usedTokenSet = useSelector(usedTokenSetSelector);
+  const themeObjects = useSelector(themeObjectsSelector);
+  const tokens = useSelector(tokensSelector);
+  const { tokenFlowButton } = useFlags();
+
+  const handleOpenTokenFlowApp = useCallback(async () => {
```
Comment (six7) [inline src/app/components/Navbar.tsx:36]: Can sou move this to a hook? Ideally all that was added, this file only needs to call 1 function in a hook

File: src/app/components/Navbar.tsx [hunk 3]
```diff
@@ -42,6 +78,7 @@ const Navbar: React.FC = () => {
         </div>
         <NavbarUndoButton />
       </Stack>
+      {tokenFlowButton && <IconButton tooltip="Open tokenflow app" dataCy="token-flow-button" onClick={handleOpenTokenFlowApp} icon={<LightningBoltIcon />} />}
```
Comment (six7) [inline src/app/components/Navbar.tsx:81]:
```suggestion
      {tokenFlowButton && <IconButton tooltip="Open visualization" dataCy="token-flow-button" onClick={handleOpenTokenFlowApp} icon={<LightningBoltIcon />} />}
```

File: src/app/store/remoteTokens.tsx [hunk 4]
```diff
@@ -98,6 +98,7 @@ export default function useRemoteTokens() {
     }
     if (remoteData?.status === 'success') {
       saveLastSyncedState(dispatch, remoteData.tokens, remoteData.themes, remoteData.metadata);
+      console.log('values', remoteData.tokens);
```
Comment (six7) [inline src/app/store/remoteTokens.tsx:101]: Log

File: src/app/components/__tests__/Navbar.test.tsx [hunk 5]
```diff
@@ -18,4 +22,12 @@ describe('Navbar', () => {
 
     expect(mockStore.getState().uiState.activeTab).toEqual(Tabs.SETTINGS);
   });
+
+  it('displays the token flow button if user is on pro plan', () => {
```
Comment (six7) [inline src/app/components/__tests__/Navbar.test.tsx:26]: Just to clarify: this is not what we want. Its a special flag that is not true on the pro plan

File: src/app/components/EditTokenForm.tsx [hunk 6]
```diff
@@ -82,11 +82,14 @@ function EditTokenForm({ resolvedTokens }: Props) {
     if ((internalEditToken?.status !== EditTokenFormStatus.EDIT || nameWasChanged) && hasNameThatExistsAlready) {
       setError('Token names must be unique');
     }
-    if ((internalEditToken?.status !== EditTokenFormStatus.EDIT || nameWasChanged) && hasAnotherTokenThatStartsWithName) {
```
Comment (six7) [inline src/app/components/EditTokenForm.tsx:85]: Generally: I see lots of changes in this PR unrelated to the functionality you added which makes it hard to review it

**Review by six7 — CHANGES_REQUESTED** [review]

File: src/app/components/Navbar.tsx [hunk 7]
```diff
@@ -8,17 +10,51 @@ import { NavbarUndoButton } from './NavbarUndoButton';
...
+      url: 'https://token-flow-app.herokuapp.com/api/tokens',
+      data: {
+        tokenData,
+        activeTheme,
+        availableThemes,
+        usedTokenSet,
+        themeObjects,
+      },
+    });
+    if (response.status === 200) window.open(`https://token-flow-app.herokuapp.com?id=${response.data.result}`);
```
Comment (roppazvan) [inline src/app/components/Navbar.tsx:49]: I think the URL should not be hardcoded, but an env variable.

File: src/app/components/Navbar.tsx [hunk 8]
```diff
@@ -8,17 +10,51 @@ import { NavbarUndoButton } from './NavbarUndoButton';
...
+  const usedTokenSet = useSelector(usedTokenSetSelector);
+  const themeObjects = useSelector(themeObjectsSelector);
+  const tokens = useSelector(tokensSelector);
+  const { tokenFlowButton } = useFlags();
+
+  const handleOpenTokenFlowApp = useCallback(async () => {
+    const tokenData = JSON.stringify(tokens, null, 2);
+    const response = await axios({
+      method: 'post',
+      url: 'https://token-flow-app.herokuapp.com/api/tokens',
```
Comment (roppazvan) [inline src/app/components/Navbar.tsx:40]: I think the URL should not be hardcoded, but an env variable.

File: src/app/components/__tests__/Navbar.test.tsx [hunk 9]
```diff
@@ -18,4 +23,22 @@ describe('Navbar', () => {
...
+  });
+
+  it('should open the token flow page when the button is clicked', async () => {
+    global.open = jest.fn();
+    const result = render(<Navbar />);
+
+    const tokenFlowButton = await result.findByTestId('token-flow-button');
+    await act(async () => userEvent.click(tokenFlowButton));
+
+    expect(global.open).toHaveBeenCalledWith('https://token-flow-app.herokuapp.com?id=test-id');
```
Comment (roppazvan) [inline src/app/components/__tests__/Navbar.test.tsx:42]: same here, please add an env variable for this.

**Review by six7 — APPROVED** [review]

**Review by six7 — APPROVED** [review]
